assets/scripts/app/Avatar.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { API_URL } from '../app/config'
+import { receiveUserDetails, hasCachedProfileImageUrl, getCachedProfileImageUrl } from '../users/profile_image_cache'
+
+const requests = {}
+
+export default class Avatar extends React.Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      backgroundImage: (this.props.userId && hasCachedProfileImageUrl(this.props.userId) ? getCachedProfileImageUrl(this.props.userId) : null)
```

Comment:
would this work?
`backgroundImage: (this.props.userId && getCachedProfileImageUrl(this.props.userId)) || null`

(Saves a function call and is a little shorter)